File: src/main/java/com/arekalov/islab1/entity/Coordinates.java
package com.arekalov.islab1.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Координаты (Entity)
 */
@Entity
@Table(name = "coordinates")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Coordinates {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull(message = "Координата X не может быть null")
    @Column(nullable = false)
    private Integer x;

    @NotNull(message = "Координата Y не может быть null")
    @Min(value = -515, message = "Координата Y должна быть больше -515")
    @Column(nullable = false)
    private Integer y;
}

File: src/main/kotlin/com/arekalov/islab1/entity/Coordinates.kt
package com.arekalov.islab1.entity

import jakarta.persistence.*
import jakarta.validation.constraints.NotNull

/**
 * Сущность координат
 */
@Entity
@Table(name = "coordinates")
data class Coordinates(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    
    @field:NotNull(message = "Координата X не может быть null")
    @Column(name = "x", nullable = false)
    val x: Int,
    
    @Column(name = "y", nullable = false)
    val y: Int
) {
    // Конструктор без параметров для JPA
    constructor() : this(null, 0, 0)
}
